packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -78,4 +95,114 @@ export class ApplicationExtension<
         const {allRoutes} = params
         return allRoutes
     }
+
+    /**
+     * Called on the server to serialize the extension data that will be sent to the client.
+     *
+     * @returns SerializedExtension - The serialized extension data.
+     * @returns null if there's no asynchronous state that needs to be serialized
+     * @throws Error if the routes have not been loaded.
+     */
+    public serialize(): SerializedExtension | null {
+        if (typeof this.getRoutesAsync === 'undefined') return null
+
+        if (this._cachedRoutes === null) {
+            throw new Error(`Routes have not been loaded. Call getRoutesAsync() before serializing`)
+        }
+
+        const serializedRoutes = this._cachedRoutes.map((route) => {
+            if (!route.componentName && !route.component) {
+                throw new Error(
+                    `Route with path "${String(
+                        route.path
+                    )}" must contain either a componentName or component to be serializable in in the ${this.getName()} extension`
+                )
+            }
+
+            // Check if the route is already serialized
+            if (route.componentName) {
+                return route
+            }
+
+            if (!route.component?.displayName) {
+                throw new Error(
+                    `Component for route with path "${String(
+                        route.path
+                    )}" is missing a displayName in the ${this.getName()} extension`
+                )
+            }
+
+            return {
+                path: route.path,
+                componentName: route.component.displayName,
+                ...(route.exact && {exact: true})
+            }
```

Comment:
This is potentially lossy conversion. It's possible that there are other properties in `route` that do not get carried over.

How about this instead?

```suggestion
            return {
                ...route,
                componentName: route.component.displayName
            }
```

Yeah, it's possible that some properties of `route` are not serializable. But I think they won't be a problem. When we [serialize the `windowGlobals`](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/12b071b35f3f8f32b04cf1833b5b864d1fb0c372/packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js#L382), the result is still going to be some string.